addons/l10n_sa_edi/models/account_edi_format.py

```diff
@@ -0,0 +1,805 @@
+        if not company._l10n_sa_check_organization_unit():
+            errors.append(
+                _("- The eleventh digit of your company VAT identification is equal to 1, in this case, the company's Organisation Unit must be a 10-digit TIN."))
+        if not company.l10n_sa_private_key:
+            errors.append(
+                _("- No Private Key was generated for company %s. A Private Key is mandatory in order to generate Certificate Signing Requests (CSR).") % company.name)
+        if not company.l10n_sa_serial_number:
+            errors.append(
+                _("- No Serial Number was assigned for company %s. A Serial Number (provided by ZATCA) is mandatory in order to generate Certificate Signing Requests (CSR).") % company.name)
+
+        supplier_missing_info = self._l10n_sa_check_partner_missing_info(
+            invoice.company_id.partner_id.commercial_partner_id)
+        customer_missing_info = self._l10n_sa_check_partner_missing_info(invoice.commercial_partner_id)
+
+        if supplier_missing_info:
+            errors.append(_set_missing_partner_fields(supplier_missing_info, _("Supplier")))
+        if customer_missing_info:
+            errors.append(_set_missing_partner_fields(customer_missing_info, _("Customer")))
+        if invoice.invoice_date > date.today():
+            errors.append(_("- Please, make sure the invoice date is set to either the same as or before Today."))
+        if invoice.move_type in ('in_refund', 'out_refund') and not (
+                (invoice.reversed_entry_id or invoice.ref) and invoice.l10n_sa_reversal_reason
+        ):
+            errors.append(
+                _("- Please, make sure both the Reversed Entry and the Reversal Reason are specified when confirming a Credit/Debit note"))
+        return errors
+
+    def _needs_web_services(self):
+        """
+            Override to add a check on edi document format code
+        """
+        self.ensure_one()
+        return self.code == 'sa_zatca' or super()._needs_web_services()
+
+    def _is_compatible_with_journal(self, journal):
+        """
+            Override to add a check on journal type & country code (SA)
+        """
+        self.ensure_one()
+        if self.code != 'sa_zatca':
+            return super()._is_compatible_with_journal(journal)
+        return journal.type == 'sale' and journal.country_code == 'SA'
+
+    def _post_invoice_edi(self, invoices):
+        """
+            Override to post ZATCA edi formats
+        """
+        self.ensure_one()
+        invoice = invoices
+        if self.code != 'sa_zatca' or invoice.company_id.country_code != 'SA':
+            return super()._post_invoice_edi(invoices)
+        if not invoice.company_id.l10n_sa_compliance_checks_passed:
+            raise UserError(_("ZATCA Compliance Checks need to be completed for the current company "
```

Comment:
In some cases some of these functions are used in other parts of the code, such as generating sample invoices to be used for the compliance checks, signing them and sending them to ZATCA, so in case of an issue, we need to show a proper error. I have added a try/except block at the root of the ZATCA submissions logic, however, which returns the proper dictionary expected

Reply:
Made it so if there are any expected errors, they will be caught and returned in the proper format expected by the EDI processing cron (dict with error/success key)